dot/core/service_test.go

```diff
@@ -28,6 +36,78 @@ import (
 
 var errTestDummyError = errors.New("test dummy error")
 
+func generateTestCentrifugeMetadata(t *testing.T) *ctypes.Metadata {
+	t.Helper()
+	rawMeta := common.MustHexToBytes(testdata.NewTestMetadata())
+	var decoded []byte
+	err := scale.Unmarshal(rawMeta, &decoded)
+	require.NoError(t, err)
+
+	meta := &ctypes.Metadata{}
+	err = ctypes.DecodeFromBytes(decoded, meta)
+	require.NoError(t, err)
+	return meta
+}
+
+func generateExtrinsic(t *testing.T) (ext types.Extrinsic, externExt types.Extrinsic, body *types.Body) {
+	t.Helper()
+	meta := generateTestCentrifugeMetadata(t)
+
+	testAPIItem := runtime.APIItem{
+		Name: [8]byte{1, 2, 3, 4, 5, 6, 7, 8},
+		Ver:  99,
+	}
+	rv := runtime.NewVersionData(
+		[]byte("polkadot"),
+		[]byte("parity-polkadot"),
+		0,
+		25,
+		0,
+		[]runtime.APIItem{testAPIItem},
+		5,
```

Comment:
nit it would be nice to have these as named inlined constants above `rv := runtime.NewVersionData(` so a reader doesn't have to lookup the method signature and match arguments

Reply:
updated, might want to double check I did what you had in mind @qdm12

Reply:
Yes that works! Although supernitting here, inlined constants just above their usage are better readability-wise since it's more 'local'.